spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -408,27 +441,26 @@ private T processInternal(ParametersWrapper parameters) throws Exception {
 			}
 			else if (e instanceof IllegalStateException) {
 				if (e.getCause() instanceof IllegalArgumentException
-						&& !"argument type mismatch".equals(e.getCause().getMessage())) {
+						&& (!"argument type mismatch".equals(e.getCause().getMessage())
+						// JVM generates GeneratedMethodAccessor### after several calls with less error checking
+						&& !e.getCause().getMessage().startsWith("java.lang.ClassCastException@"))) {
 					throw e;
 				}
 			}
```

Comment:
I think there are still some issues with this exception logic and falling back to SpEL.

If the method was invoked correctly but the user code threw the exception (e.g `IllegalStateException`) we'd retry with SpEL (and we shouldn't).

We need to (somehow) more reliably detect that the exception was thrown by the `InvocableHandlerMethod` itself.